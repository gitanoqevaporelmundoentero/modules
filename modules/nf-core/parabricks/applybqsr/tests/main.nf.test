nextflow_process {

    name "Test Process PARABRICKS_APPLYBQSR"
    script "../main.nf"
    process "PARABRICKS_APPLYBQSR"

    tag "modules"
    tag "modules_nfcore"
    tag "parabricks"
    tag "parabricks/applybqsr"

    test("parabricks_applybqsr") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true)
                ]
                input[1] = [
                    [], []  // index not needed unless using intervals
                ]
                input[2] = [
                    [ id:'bqsr_table'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/gatk/test.baserecalibrator.table', checkIfExists: true)
                ]
                input[3] = [
                    [], []  // no intervals
                ]
                input[4] = [
                    [ id:'reference'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    bam(process.out.bam[0][1]).getReadsMD5(),
                    file(process.out.bai[0][1]).name,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("parabricks_applybqsr_intervals") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.single_end.sorted.bam', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test-index'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.single_end.sorted.bam.bai', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'bqsr_table'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/gatk/test.baserecalibrator.table', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'intervals'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'reference'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                   ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
                    bam(process.out.bam[0][1]).getReadsMD5(),
                    file(process.out.bai[0][1]).name,
                    process.out.versions
                ).match() }
            )
        }

    }

    test("parabricks_applybqsr_intervals_stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.single_end.sorted.bam', checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test-index'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.single_end.sorted.bam.bai', checkIfExists: true)
                ]
                input[2] = [
                    [ id:'bqsr_table'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/gatk/test.baserecalibrator.table', checkIfExists: true)
                ]
                input[3] = [
                    [ id:'intervals'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/bed/test.bed', checkIfExists: true)
                ]
                input[4] = [
                    [ id:'reference'],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                   ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
