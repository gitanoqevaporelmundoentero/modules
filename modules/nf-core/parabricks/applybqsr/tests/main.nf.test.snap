{
    "parabricks_applybqsr_intervals": {
        "content": [
            "d41d8cd98f00b204e9800998ecf8427e",
            "test.bam.bai",
            [
                "versions.yml:md5,139b8f2a0926a26cf843ac23daace6d0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T10:41:06.541345938"
    },
    "parabricks_applybqsr": {
        "content": [
            "894549ee3ced6b5ca2eed2563a985217",
            "test.bam.bai",
            [
                "versions.yml:md5,139b8f2a0926a26cf843ac23daace6d0"
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T10:40:59.520632861"
    },
    "parabricks_applybqsr_intervals_stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,139b8f2a0926a26cf843ac23daace6d0"
                ],
                "bai": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam.bai:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test"
                        },
                        "test.bam:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,139b8f2a0926a26cf843ac23daace6d0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-10-30T10:41:12.314674351"
    }
}